crates/host/src/wasmbus/mod.rs

```diff
@@ -1750,7 +1751,9 @@ impl Host {
                     max_instances,
                     &annotations,
                     config,
-                    wasm,
+                    wasm.ok_or_else(|| {
+                        retrieval_error.unwrap_or(anyhow!("unexpected retrieval error"))
```

Comment:
```suggestion
                        retrieval_error.unwrap_or_else(|| anyhow!("unexpected retrieval error"))
```

What does it mean if there is no retrieval error, but the wasm is missing? Unless I'm misunderstanding I think the message here isn't quite right

Reply:
I don't think that path is actually possible right now, maybe something like 'unexpected missing wasm'?